src/NzbDrone.Core/Download/Clients/Flood/FloodSettings.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation;
+using NzbDrone.Core.Annotations;
+using NzbDrone.Core.Download.Clients.Flood.Models;
+using NzbDrone.Core.ThingiProvider;
+using NzbDrone.Core.Validation;
+
+namespace NzbDrone.Core.Download.Clients.Flood
+{
+    public class FloodSettingsValidator : AbstractValidator<FloodSettings>
+    {
+        public FloodSettingsValidator()
+        {
+            RuleFor(c => c.Host).ValidHost();
+            RuleFor(c => c.Port).InclusiveBetween(1, 65535);
+        }
+    }
+
+    public class FloodSettings : IProviderConfig
+    {
+        private static readonly FloodSettingsValidator Validator = new FloodSettingsValidator();
+
+        public FloodSettings()
+        {
+            UseSsl = false;
+            Host = "localhost";
+            Port = 3000;
+            Tags = new string[]
+            {
+                "radarr"
+            };
+            AdditionalTags = Enumerable.Empty<int>();
+            StartOnAdd = true;
+        }
+
+        [FieldDefinition(0, Label = "Use SSL", Type = FieldType.Checkbox)]
+        public bool UseSsl { get; set; }
+
+        [FieldDefinition(1, Label = "Host", Type = FieldType.Textbox)]
+        public string Host { get; set; }
+
+        [FieldDefinition(2, Label = "Port", Type = FieldType.Textbox)]
+        public int Port { get; set; }
+
+        [FieldDefinition(3, Label = "Url Base", Type = FieldType.Textbox, HelpText = "Optionally adds a prefix to Flood API, such as [protocol]://[host]:[port]/[urlBase]api")]
+        public string UrlBase { get; set; }
+
+        [FieldDefinition(4, Label = "Username", Type = FieldType.Textbox, Privacy = PrivacyLevel.UserName)]
+        public string Username { get; set; }
+
+        [FieldDefinition(5, Label = "Password", Type = FieldType.Password, Privacy = PrivacyLevel.Password)]
+        public string Password { get; set; }
+
+        [FieldDefinition(6, Label = "Destination", Type = FieldType.Textbox, HelpText = "Manually specifies download destination")]
+        public string Destination { get; set; }
+
+        [FieldDefinition(7, Label = "Tags", Type = FieldType.Tag, HelpText = "Initial tags of a download. To be recognized, a download must have all initial tags. This avoids conflicts with unrelated downloads.")]
+        public IEnumerable<string> Tags { get; set; }
+
+        [FieldDefinition(8, Label = "Post-Import Tags", Type = FieldType.Tag, HelpText = "Appends tags after a download is imported.", Advanced = true)]
+        public IEnumerable<string> PostImportTags { get; set; }
+
+        [FieldDefinition(9, Label = "Additional Tags", Type = FieldType.Select, SelectOptions = typeof(AdditionalTags), HelpText = "Adds properties of media as tags. Hints are examples.", Advanced = true)]
+        public IEnumerable<int> AdditionalTags { get; set; }
+
+        [FieldDefinition(10, Label = "Start on Add", Type = FieldType.Checkbox)]
```

Comment:
All the other clients use a setting "Add Paused" defaulted to false (same basic functionality), can we switch this around to be consistent?

Reply:
Flood's API uses `start`. Additionally, this is already in Sonarr. It is probably better to keep it as it is.

Reply:
The Flood API doesn't need to change, logic in the proxy can handle that. 

I assume Sonarr just missed it on review, but I'm sure they would take a PR to fix it up for consistency sake. I don't think it's a good idea to have it different than every other client in Radarr/Sonarr from a user perspective.

Reply:
It would require a database migration now to change this one.

Reply:
Otherwise users will notice their settings to be unexpected flipped to the opposite site. I wouldn’t prefer that.

Reply:
The migration on Sonarr will be a call for @markus101, but I'm sure he'd agree here on the call for consistency and be open to a quick migration for this if he deems it necessary. Regardless I'd like to see this changed for merge into Radarr. 

I'm not asking you to change the default behavior, your 'StartOnAdd' defaulted to 'true' will turn into 'AddPaused' defaulted to 'false'. In either case they perform the same in the default setup. This is the expected format and behavior that both user bases currently expect and changing one client to be completely opposite adds unnecessary confusion.

Reply:
It is implemented in https://github.com/Radarr/Radarr/pull/5552/commits/9b5672ce921be78a3d82274e390f600daecfeae8. Note that I totally disagree with the change. Existing client implementation shows that the option is oriented to the APIs of respective clients. For instance, qBittorrent has `InitialState` instead while rTorrent/ruTorrent has nothing. All existing clients that uses `AddPaused` has a corresponding API interface that uses `paused` as well. This change makes the implementation contradictory to Flood's API.

Reply:
Thanks, rTorrent has `AddStopped`, and yes Deluge/NzbGet/Vuze/Transmission all use AddPaused..

To your point I wouldn't be opposed centralizing them all to a InitialState field sometime in the future, as it seems a bit cleaner. If you want to go in with that on Flood I'm fine going that route, otherwise I'll merge.

Reply:
You may proceed. There is no such plan at the moment. Flood's `isCompleted` covers such case well, and it is much more concise than `ForceStart` initial state from API point of view. My disappointment with the divergence from Flood's APIs stands.